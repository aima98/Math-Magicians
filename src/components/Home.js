export default function Home() {
  return (
    <section className="message">
      <h2>Welcome to our page!</h2>
      <p>
        This is Math Magician.
        Here, we offer a variety of tools and collection of random quotes to keep you entertained.
        Our easy-to-use calculators are designed to help you with plenty of mathematics operations.
        You are welcomed to explore by yourself all what our website has to offer.
      </p>
    </section>
  );
}
