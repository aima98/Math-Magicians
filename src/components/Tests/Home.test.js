import { render, screen } from "@testing-library/react";
import renderer from 'react-test-renderer';
import Home from "../Home";

describe('render tests', () => {
  test('render test header', () => {
    render(<Home />);
    const welcomeHeader = screen.getByRole("heading");
    expect(welcomeHeader.innerHTML).toBe('Welcome to our page!');
  });

  test('render test paragraph', () => {
    render(<Home />);
    const paragraph = screen.queryByText(/This is Math Magician./i);
    expect(paragraph).toHaveTextContent("Here, we offer a variety of tools and collection of random quotes to keep you entertained.");
  });
});

describe('snapshot test for Home', () => {
  test('snapshot test', () => {
    const homeTree = renderer.create(<Home />).toJSON();
    expect(homeTree).toMatchSnapshot();
  });
});