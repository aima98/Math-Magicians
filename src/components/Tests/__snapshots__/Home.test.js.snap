// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for Home snapshot test 1`] = `
<section
  className="message"
>
  <h2>
    Welcome to our page!
  </h2>
  <p>
    This is Math Magician. Here, we offer a variety of tools and collection of random quotes to keep you entertained. Our easy-to-use calculators are designed to help you with plenty of mathematics operations. You are welcomed to explore by yourself all what our website has to offer.
  </p>
</section>
`;
